### ISSUE TITLE
UnusedImports does not report unused static imports when the method is used as a method reference

### ISSUE THREAD

**Issue by ksobolew** [checkstyle/checkstyle#12923]
I have read check documentation: https://checkstyle.sourceforge.io/config_imports.html#UnusedImports
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
kudivuhadi@MBP-Krzysztof scratches %  javac scratch.java 
kudivuhadi@MBP-Krzysztof scratches %  ls -l
total 24
-rw-r--r--  1 kudivuhadi  staff  1165 28 mar 12:00 Dummy.class
-rw-r--r--@ 1 kudivuhadi  staff   223 28 mar 12:00 scratch.java

kudivuhadi@MBP-Krzysztof scratches %  cat config.xml 
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://checkstyle.sourceforge.net/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="UnusedImports" />
    </module>
</module>

kudivuhadi@MBP-Krzysztof scratches %  cat scratch.java
import java.util.Optional;
import java.util.Set;

import static java.lang.String.format;
import static java.lang.String.join;

class Dummy
{
    public static void main(String[] args)
    {
        Optional<String> test = Optional.empty();
        test.map(String::format);
    }
}

kudivuhadi@MBP-Krzysztof scratches %  RUN_LOCALE="-Duser.language=en -Duser.country=US"
kudivuhadi@MBP-Krzysztof scratches %  java $RUN_LOCALE -jar checkstyle-10.9.3-all.jar \
   -c config.xml scratch.java
Starting audit...
[ERROR] scratch.java:2:8: Unused import - java.util.Set. [UnusedImports]
[ERROR] scratch.java:5:15: Unused import - java.lang.String.join. [UnusedImports]
Audit done.
Checkstyle ends with 2 errors.
```

---

The Java file contains three unused imports: the import of `java.util.Set` and static imports of `java.lang.String#format` and `java.lang.String#join`. Only two of them are reported, while the static import of `java.lang.String#format` is a false negative. From what it looks like, it's not reported because it's "used" as a method reference, as in `test.map(String::format);`, which is wrong, because with a method reference only the type needs to be imported and that's enough to resolve the method itself.

**Comment by nrmancuso** [comment]
@ksobolew interesting nuance, thanks for raising this issue!

**Comment by 0xbakry** [comment]
@nrmancuso can i work on this?

**Comment by nrmancuso** [comment]
@0xbakry we have a PR in progress, please select another issue

**Comment by romani** [comment]
@mahfouz72 , can you help us to finish this issue ? fix is almost done at https://github.com/checkstyle/checkstyle/pull/12933, only small step is required. please create new PR and resuse code

**Comment by mahfouz72** [comment]
@romani ok will take it.will send a new PR asap

**Comment by mahfouz72** [comment]
@romani I think this should be closed now
